cloud/scope/cluster.go

```diff
@@ -83,11 +84,100 @@ type ClusterScope struct {
 	AzureCluster *infrav1.AzureCluster
 }
 
+// NewMachineScope creates a new MachineScope from the supplied parameters.
+// This is meant to be called for each reconcile iteration.
+func (s *ClusterScope) NewMachineScope(params MachineScopeParams) (*MachineScope, error) {
```

Comment:
nit: it makes me twitch that we have the constructor of the other object receiver like this instead of with the rest of machine scope? can we not pass cluster scope into  NewMachineScope()

Reply:
Definitely could pass in the cluster scope. It is effectively the same thing.

Reply:
yeah the two would be equivalent... I guess it felt cleaner to have `func (s *ClusterScope) NewMachineScope(params MachineScopeParams) (*MachineScope, error) {` than `func NewMachineScope(s *ClusterScope, params MachineScopeParams) (*MachineScope, error) {` and this was a way to be very explicit about the dependency of MachineScope on ClusterScope 

(same thing applies to MachinePool)

Reply:
add to param struct?

Reply:
yeah that's good feedback, i agree the constructor isn't really in the right place right now. Let me try that.

Reply:
done